privacyidea/api/user.py

```diff
@@ -104,14 +105,96 @@ def get_users():
         }
     """
     realm = getParam(request.all_data, "realm")
-    users = get_user_list(request.all_data)
+    attr = is_attribute_at_all()
+    users = get_user_list(request.all_data, custom_attributes=attr)
 
     g.audit_object.log({'success': True,
                         'info': "realm: {0!s}".format(realm)})
     
     return send_result(users)
 
 
+@user_blueprint.route('/attribute', methods=['POST'])
+@prepolicy(check_custom_user_attributes, request, "set")
+@user_required
+@event("set_custom_user_attribute", request, g)
+def set_user_attribute():
+    """
+    Set a custom attribute for an external user.
+    The user is specified by the usual parameters user, resolver and realm.
+
+    :httpparam key: The name of the attributes
+    :httpparam value: The value of the attribute
+    :httpparam type: an optional type of the attribute
+
+    The database id of the attribute is returned. The return
+    value thus should be >=0.
+    """
+    # We basically need a user, otherwise we will fail, but the
+    # user object is later simply used from request.User. We only
+    # need to avoid an empty User object.
+    _user = getParam(request.all_data, "user", optional=False)
+    attrkey = getParam(request.all_data, "key", optional=False)
+    attrvalue = getParam(request.all_data, "value", optional=False)
+    attrtype = getParam(request.all_data, "type", optional=True)
+    r = request.User.set_attribute(attrkey, attrvalue, attrtype)
+    return send_result(r)
+
+
+@user_blueprint.route('/attribute', methods=['GET'])
+@user_required
+@event("get_user_attribute", request, g)
+def get_user_attribute():
+    """
+    Return the custom attribute of the given user.
+    The user is specified by the usual parameters user, resolver and realm.
```

Comment:
I think we should stress out that this endpoint only returns the custom attributes, not the ones from the userstore. All attributes are available in the ``/user/`` endpoint.

Reply:
Done in 3513c660